tests/network/libs/cluster_user_defined_network.py

```diff
@@ -0,0 +1,113 @@
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any
+
+from ocp_resources.cluster_user_defined_network import ClusterUserDefinedNetwork as Cudn
+from timeout_sampler import retry
+
+from tests.network.libs.apimachinery import dict_normalization_for_dataclass
+
+_WAIT_FOR_STATUS_TIMEOUT_SEC = 90
+_WAIT_FOR_STATUS_INTERVAL_SEC = 5
+
+
+@dataclass
+class NamespaceSelector:
+    matchLabels: dict[str, str]  # noqa: N815
+
+
+@dataclass
+class Access:
+    id: int
+
+
+@dataclass
+class Vlan:
+    class Mode(Enum):
+        ACCESS = "Access"
+
+    access: Access
+    mode: str
+
+
+@dataclass
+class Ipam:
+    class Mode(Enum):
+        DISABLED = "Disabled"
+
+    mode: str
+
+
+@dataclass
+class Localnet:
+    class Role(Enum):
+        SECONDARY = "Secondary"
+
+    role: str
+    physicalNetworkName: str  # noqa: N815
+    vlan: Vlan
+    ipam: Ipam
+
+
+@dataclass
+class Network:
+    class Topology(Enum):
+        LOCALNET = "Localnet"
+
+    topology: str
+    localnet: Localnet
+
+
+class ClusterUserDefinedNetwork(Cudn):
+    """
+    ClusterUserDefinedNetwork object.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        namespace_selector: NamespaceSelector,
+        network: Network,
+    ):
+        """
+        Create and manage ClusterUserDefinedNetwork
+
+        API reference:
+        https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#clusteruserdefinednetwork
+
+        Args:
+            name (str): Name of the ClusterUserDefinedNetwork object.
+            namespace_selector (NamespaceSelector): NamespaceSelector Label selector for which namespace network should
+                be available for.
+            network (Network): Network is the user-defined-network spec.
+        """
+        self._namespace_selector = namespace_selector
+        self._network = network
+        super().__init__(
+            name=name,
+            namespace_selector=asdict(namespace_selector, dict_factory=dict_normalization_for_dataclass),
+            network=asdict(network, dict_factory=dict_normalization_for_dataclass),
+        )
+
+    class Status:
+        class Condition:
+            class Type(str, Enum):
+                NETWORK_CREATED = "NetworkCreated"
+
+    @retry(
+        wait_timeout=_WAIT_FOR_STATUS_TIMEOUT_SEC,
```

Comment:
use directly TIMEOUT constants from utilities/constants.py

Reply:
sig-network decided not to use them anymore.

Reply:
not in this PR but we need to talk about this